### ISSUE TITLE
As a user, I want deep archive to support LBLX label extensions

### ISSUE THREAD

**Issue by tbarnes4** [NASA-PDS/deep-archive#157]
### Checked for duplicates

No - I haven't checked

### 🐛 Describe the bug

When I run the software on a bundle that has only lblx files, the software runs without errors, but produces empty SIP, checksum, and manifest *.tab files.

### 🕵️ Expected behavior

I expect the software to create the tables as if the bundle contained xml labels.

### 📜 To Reproduce

1. Run pds-deep-archive 
2. Ex: pds-deep-archive -s PDS_SBN -b https://pdssbn.astro.umd.edu/holdings/ --include-latest-collection-only pds4-nh_documents-v1.0/bundle.lblx
3.
...

### 🖥 Environment Info

- Version of this software 1.1.4
- Operating System: MacOSX 12.6.8 and Red Hat Enterprise Linux 9.3
...

### 📚 Version of Software Used

pds-deep-archive 1.1.4

### 🩺 Test Data / Additional context

https://pdssbn.astro.umd.edu/holdings/pds4-nh_documents-v1.0

### 🦄 Related requirements

🦄 #xyz

**Comment by tbarnes4** [comment]
@jordanpadams Since I've seen no further comment on this, just pinging again.

**Comment by jordanpadams** [comment]
@tbarnes4 thanks for the ping. I updated this ticket to be a new requirement since this is an overall new requirement on the system that was not implemented yet.

That being said, we added this to the backlog, but it is not on the top of our radar since there is a workaround.

If you load this data into the Registry/API (which it looks like it is), then you can run `pds-deep-registry-archive` to generate the SIPs much easier (and faster), not to mention it is a good forcing factor to load the data into the registry.

```
$ pds-deep-registry-archive --site PDS_SBN urn:nasa:pds:nh_documents::1.0
```

**Comment by jordanpadams** [comment]
@tbarnes4 actually... standby on running deep-archive. I forgot we have a bug right now. A PR is up and we will hopefully have a new version tagged by end of day (or first thing Monday).

**Comment by jordanpadams** [comment]
@tbarnes4 you should be able to [upgrade your PDS Deep Archive package](https://nasa-pds.github.io/deep-archive/installation/index.html#upgrading-the-software) and run the utility as described above. I just tested it and it appears to work great.

**Comment by tbarnes4** [comment]
@jordanpadams Upgraded to v1.1.15 on two different systems (same as above) and I am still getting the same result with the same bundle (as above) where the 3 *tab files are all empty.

> % source ~/.virtualenvs/pds-deep-archive/bin/activate.csh
> % pds-deep-archive -s PDS_SBN -b https://pdssbn.astro.umd.edu/holdings/ pds4-nh_documents-v1.0/bundle.lblx
> INFO 👟 PDS Deep Archive, version 1.1.5
> INFO 🏃‍♀️ Starting AIP generation for pds4-nh_documents-v1.0/bundle.lblx
> INFO 🎉 Success! AIP done, files generated:
> INFO 📄 Checksum manifest: nh_documents_v1.0_20240227_checksum_manifest_v1.0.tab
> INFO 📄 Transfer manifest: nh_documents_v1.0_20240227_transfer_manifest_v1.0.tab
> INFO 📄 XML label for them both: nh_documents_v1.0_20240227_aip_v1.0.xml
> INFO 🏃‍♀️ Starting SIP generation for pds4-nh_documents-v1.0/bundle.lblx
> INFO 🎉 Success! From /Volumes/echeclus02/TEMP/nssdc_test/pds4-nh_documents-v1.0/bundle.lblx, generated these output files:
> INFO 📄 SIP Manifest: nh_documents_v1.0_20240227_sip_v1.0.tab
> INFO 📄 XML label for the SIP: nh_documents_v1.0_20240227_sip_v1.0.xml
> INFO 👋 That's it for now. Bye.

**Comment by jordanpadams** [comment]
@tbarnes4 sorry for the confusion, I meant you are good to use the `pds-deep-registry-archive` utility to generate the SIPs using the next-gen Registry. This requirement has still not been implemented, and will unfortunately not be implemented for a bit since we have a known workaround.

**Comment by tbarnes4** [comment]
@jordanpadams That makes much more sense.  Tracking down other issues and I felt I was going crazy.   I expect to try the registry-archive later.

**Comment by jordanpadams** [comment]
@tbarnes4 copy that. it should take ~30 seconds to try it :-) . and you can run it from anywhere (your own laptop works).

**Comment by tbarnes4** [comment]
@jordanpadams That it does, even less.  For kicks and giggles tried a collection LIDVID and it errored out with unexpected errors.  But can't expect it to do things it wasn't designed to do.  ;-)

**Comment by smclaughlin7** [comment]
Subscribed to monitor progress. @tbarnes4 Do you expect to submit this bundle shortly after deep archive is enhanced?

**Comment by smclaughlin7** [comment]
@jordanpadams  Should NSSDCA expect Deep Archive tool continue to use '.xml' as the extension for SIP products, at least until IM 2.0?  I assume '.lblx' is not presently an option for that tool, correct?  Thanks!

**Comment by jordanpadams** [comment]
@smclaughlin7 correct. And I am not even sure we will be upgrading with IM 2.0 since all of these SIPs belong to the same collection. We don't plan on going back and reprocessing those SIPs to use lblx.

**Comment by smclaughlin7** [comment]
@jordanpadams Thanks! We/NSSDCA will continue to expect only '.xml' for SIPs.

**Comment by tbarnes4** [comment]
> Subscribed to monitor progress. @tbarnes4 Do you expect to submit this bundle shortly after deep archive is enhanced?

@smclaughlin7 Sorry for the delay.  I have just [submitted](https://github.com/NASA-PDS/operations/issues/507) the two bundles.  Hopefully they go thru without problem.  Thanks!

**Comment by tbarnes4** [comment]
@jordanpadams I think this needs to be addressed sooner.  I have been sitting on data sets in hopes to register them and then using the pds-deep-registry-archive, but it has been months since we have been able to register new data.  I have submitted my non-lblx data, but I am unable to submit my data sets that utilize lblx label file extensions.  Is there a quick fix?  It is not good for us to sit on ready to deep archive data.  Thanks.

**Comment by jordanpadams** [comment]
@tbarnes4 apologies for the delay, but we have no available staff to work on this at this time, as we have been focused on getting the Registry up and running, which was completed last week. Pat should be receiving info to again begin loading data into the registry. Sorry for the inconvenience.

**Comment by jordanpadams** [comment]
Closed per https://github.com/NASA-PDS/deep-archive/pull/205

**Comment by jordanpadams** [comment]
thanks to @nutjob4life for getting this completed so quickly! 🎉

**Comment by tbarnes4** [comment]
@jordanpadams Thanks! Perhaps I'm missing something, but when I install the
development version (to keep seperate from my stable version) using the
lines you gave, it says it is version 1.1.2, though the current stable
version is 1.3.0, let alone what I would expect the development version to
be.  I saw an earlier email from PDSEN CI Bot saying 1.4.0-dev was
released.  Hopefully this is just an ignorance question on my part.  Thanks
again for your team quickly tackling this.

**Comment by nutjob4life** [comment]
@tbarnes4 yeah, not sure what happened, but the version on test.pypi.org is still 4 years old.

For now, you can try installing from source. Replace the `pip3` command [in Jordan's comment](https://github.com/NASA-PDS/deep-archive/issues/157#issuecomment-2790237600) with this:
```console
pip3 install git+https://github.com/NASA-PDS/deep-archive.git@main
```
No need to uninstall the old pds.deeparchive; this command will overwrite it. 

You should then see:
```
$ pds-deep-registry-archive --version
pds-deep-registry-archive 1.4.0
```

**Comment by tbarnes4** [comment]
@*nutjob4life *Got it, thanks!

Doing a quick test, the update for lblx file support appears to work for
pds-deep-archive.

**Comment by jordanpadams** [comment]
great thanks @tbarnes4 ! we will try to tag a release soon.
